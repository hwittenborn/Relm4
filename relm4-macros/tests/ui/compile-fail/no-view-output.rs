use relm4::prelude::*;

struct AppModel;

#[relm4_macros::component]
impl SimpleComponent for AppModel {
    type Init = ();
    type Input = ();
    type Output = ();

    view! {
        gtk::Window {}
    }

    #[allow(unreachable_code)]
    fn init(
        _: (),
        root: Self::Root,
        sender: ComponentSender<Self>,
    ) -> ComponentParts<Self> {
        let model = Self;

        let widgets = todo!();

        ComponentParts { model, widgets }
    }
}

fn main() {}
