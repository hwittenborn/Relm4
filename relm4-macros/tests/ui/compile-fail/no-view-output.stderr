error: expected an injection point for the view macro. Try using `let widgets = view_output!();`
  --> tests/ui/compile-fail/no-view-output.rs:15:5
   |
15 | /     #[allow(unreachable_code)]
16 | |     fn init(
17 | |         _: (),
18 | |         root: Self::Root,
...  |
25 | |         ComponentParts { model, widgets }
26 | |     }
   | |_____^
